web/src/store/index.ts

```diff
@@ -94,6 +106,45 @@ const {
           const activeDataset = tileSourceDatasets.length > 0 ? tileSourceDatasets[0] : null;
           commit.setActiveDataset(activeDataset);
 
+          const embeddings: DatasetEmbedding[] = (await store.state.axiosInstance.get(`/investigations/${investigationId}/embeddings`)).data;
+          commit.setDatasetEmbeddings(embeddings);
+
+          const metadataPromises: Promise<{ datasetId: string; result: AxiosResponse }>[] = [];
+          tileSourceDatasets.forEach((dataset) => {
+            const promise = store.state.axiosInstance?.get(`/datasets/${dataset.id}/tiles/metadata`).then((result) => ({
+              datasetId: dataset.id,
+              result,
+            }));
+            if (promise) {
+              metadataPromises.push(promise);
+            }
+          });
+          embeddings.forEach((embedding) => {
+            const promise = store.state.axiosInstance?.get(`/datasets/${embedding.child}/tiles/metadata`).then((result) => ({
+              datasetId: embedding.child,
+              result,
+            }));
+            if (promise) {
+              metadataPromises.push(promise);
+            }
+          });
+          embeddings.forEach((embedding) => {
+            const promise = store.state.axiosInstance?.get(`/datasets/${embedding.parent}/tiles/metadata`).then((result) => ({
+              datasetId: embedding.parent,
+              result,
+            }));
+            if (promise) {
+              metadataPromises.push(promise);
+            }
+          });
```

Comment:
Looks like there's some duplicated code here.

Reply:
Just kidding.

Reply:
Just bad code. One does `embedding.parent` and the other does `embedding.child`.